src/main/scala/ai/privado/utility/ConcurrentProcessor.scala

```diff
@@ -39,13 +40,19 @@ object ConcurrentProcessor {
   * @tparam R
   *   \- Type of the result object
   */
-abstract class ConcurrentProcessor[T, R](result: R) {
+abstract class ConcurrentProcessor[T, R](result: R, rules: RuleCache) {
   private val logger = LoggerFactory.getLogger(this.getClass)
   // PUSH socket maintained for each task thread (Future)
   private val queue: ThreadLocal[ZMQ.Socket] = new ThreadLocal[ZMQ.Socket]
   // Unique ZeroMQ inproc:// topic used for each instance of ConcurrentProcessor.
   private val topic: String = UUID.randomUUID().toString
 
+  // set maximum sockets that can be used
+  private val count          = rules.getSystemConfigByKey(Constants.MAX_SOCKET_COUNT)
+  private val maxSocketCount = if (count.isEmpty) 1024 else if (count.toInt > 4000) 1024 else count.toInt
+  println(maxSocketCount)
```

Comment:
@tuxology Can we have a little descriptive message along with socket count to understand this on console?

Reply:
Nevermind was refering a older commit